src/actions/sso/paymentRequest.ts

```diff
@@ -34,11 +36,25 @@ export const consumePaymentRequest = (
     registry as JolocomRegistry,
   )
 
+  const { did: offerorDid, publicProfile } = await registry.resolve(
+    keyIdToDid(paymentRequest.issuer),
+  )
+
+  const parsedProfile = publicProfile
```

Comment:
In a future PR this can be abstracted to a utility function (given that this code will be required whenever we process an interaction request), something like:

```typescript
const extractPublicProfile = (requester: Identity) => {
  return requester.publicProfile ?
    ? omit(['id', 'did'], publicProfile.toJSON().claim)
    : {} // might make sense to return a fallback / default profile here (i.e. generic icon, did, no name provided)
}
```

Reply:
IMHO having a fallback profile here would be the responsibility of the UI and thus be handle in the component